### PR TITLE
INTMDB-456 custom db role

### PR DESCRIPTION
**Pull request by Adelmar92**
## Description

Added Custom db Role endpoint Resource.

How to test:
Can be manually executed by using this json inputs and the command:

`cfn invoke resource {CREATE | READ | UPDATE | DELETE | LIST} file-name.json`
    
CREATE
```json
{
  "desiredResourceState": {
    {
  "GroupId": {GROUP-ID},
  "Actions": [
    {
      "Action": "FIND",
      "Resources": [
        {
          "Collection": {TEST-COLLECTION},
          "DB": {TEST-DB}
        }
      ]
    }
  ],
  "RoleName": "testcreate",
  "ApiKeys": {
    "PublicKey": {PUBLIC-KEY},
    "PrivateKey": {PRIVATE-KEY}
  }
}
  },
  "previousResourceState": {
  }
}
```
    
READ
```json
{
  "desiredResourceState": {
    {
    "GroupId": {GROUP-ID},
    "RoleName": "testcreate",
    "ApiKeys": {
      "PublicKey": {PUBLIC-KEY},
      "PrivateKey": {PRIVATE-KEY}
    }
  }
  },
  "previousResourceState": {
  }
}
```

DELETE
```json
{
  "desiredResourceState": {
        {
    "GroupId": {GROUP-ID},
    "RoleName": "testcreate",
    "ApiKeys": {
      "PublicKey": {PUBLIC-KEY},
      "PrivateKey": {PRIVATE-KEY}
    }
  },
  "previousResourceState": {
  }
}
```

LIST
```json
{
  "desiredResourceState": {
   {
    "GroupId": {GROUP-ID},
    "ApiKeys": {
      "PublicKey": {PUBLIC-KEY},
      "PrivateKey": {PRIVATE-KEY}
    }
  },
  "previousResourceState": {
  }
}
```

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

### CODE REVIEW

**Comment by matt-condon** [comment]
Can we add a test for list and additionally read/list tests where no custom groups exist?

**Comment by Adelmar92** [comment]
> Can we add a test for list and additionally read/list tests where no custom groups exist?

Hi Mat! the cfn test with the inputs_1_create.template.json in the test file, already cover those cases, 
i think the case you suggest are covered in the contract_delete_read and the contract_delete_list (https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/contract-tests.html)
image

unfortunately there are no way to add unit testing to test this cases, mostly because we cannot mock the client in each resource. How do you suggest to procede?

**Comment by Zuhairahmed** [comment]
> > Can we add a test for list and additionally read/list tests where no custom groups exist?
> 
> Hi Mat! the cfn test with the inputs_1_create.template.json in the test file, already cover those cases, i think the case you suggest are covered in the contract_delete_read and the contract_delete_list (https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/contract-tests.html) image
> 
> unfortunately there are no way to add unit testing to test this cases, mostly because we cannot mock the client in each resource. How do you suggest to procede?

@matt-condon any feedback on this?

File: cfn-resources/custom-db-role/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+Both CREATE & DELETE tests must pass.
+
+## Installation
+TAGS=logging make
+cfn submit --verbose --set-default
+
+## Usage
+
+The [/quickstart-mongodb-atlas/scripts/launch-x-quickstart.sh](launch-x-quickstart.sh) script
```
Comment (Zuhairahmed) [inline cfn-resources/custom-db-role/README.md:36]: link does not work, can you fix it please?

File: cfn-resources/custom-db-role/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+```bash
+repo_root=$(git rev-parse --show-toplevel)
+source <(${repo_root}/quickstart-mongodb-atlas/scripts/export-mongocli-config.py)
+${repo_root}/quickstart-mongodb-atlas/scripts/launch-x-quickstart.sh ${repo_root}/cfn-resources/project/test/project.sample-template.yaml SampleProject1 ParameterKey=OrgId,ParameterValue=${ATLAS_ORG_ID}
+```
+
+## For More Information
+See the MongoDB Atlas API [Project Endpoint](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Projects) Documentation.
```
Comment (Zuhairahmed) [inline cfn-resources/custom-db-role/README.md:59]: Incorrect Resource Name/Link, please update

File: cfn-resources/custom-db-role/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Returns, adds, edits, and removes custom database user privilege roles.
+
+## Attributes & Parameters
+
+Please consult the [Resource Docs](docs/README.md)
+
+## Local Testing
+
+The local tests are integrated with the AWS `sam local` and `cfn invoke` tooling features:
+
```
Comment (Zuhairahmed) [inline cfn-resources/custom-db-role/README.md:12]: Missing section + link to Examples folder so users know to leverage this resource example 

https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples

**Review by Zuhairahmed — COMMENTED** [review]
Thanks @Adelmar92! Few comments below:

**Review by ParthasarathyVaradhan — APPROVED** [review]
LGTM

**Review by Zuhairahmed — APPROVED** [review]
@Adelmar92 thanks for addressing feedback. LGTM!

**Pull request merged**